src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -702,11 +702,13 @@ public bool IsReadOnly
         public void Assign(string expression)
         {
             this.VerifyNotDisposed();
+            int threadId = Client.GetDebuggedThread().Id;
+            uint frameLevel = _ctx.Level;
```

Comment:
I agree that failure is fine, as long as it fails cleanly and doesn't result in a crash or anything like that.